### PR TITLE
#311 assigner and reviewer actions in list

### LINKED ISSUE #311: Deal with users that have more than on roles: Assigner and others

### PR DESCRIPTION
**Pull request by nmadruga**
Fixes #311

### Back-end branch
[#311F](https://github.com/openmsupply/application-manager-server/pull/302)

### Based on PR
#469 

### Changes
- Update `getApplicationList.query` to also get assigner fields
- Send `userId` on useListApplications` and `getApplicationList.query` which will be checked against `reviewerId` and `assignerId` (on back-end).
- Fix problem in `ReviewerActionCell` when copying `demo-22-03` branch that caused **Assign** action to display for wrong applications, now uses the `assignCount` field and only shows when the application can be assigned and the current user is an Assigner
- Rename old user-role `reviewer1` to `reviewer` and remove `reviewer2` to simplify user-roles
- Found a new bug related to selected user-role on URL not getting updated base on current user (#603)

### CODE REVIEW

**Comment by nmadruga** [comment]
Hi @CarlosNZ and @andreievg can some of you have a look on this PR please? It's 7 days old now.

File: src/utils/data/columnsPerUserRole.ts [hunk 1]
```diff
@@ -13,23 +13,15 @@ const COLUMNS_PER_ROLE: ColumnsPerRole = {
...
-  reviewer2: [
-    APPLICATION_COLUMNS.SERIAL_NUMBER,
-    APPLICATION_COLUMNS.LAST_ACTIVE_DATE,
-    APPLICATION_COLUMNS.APPLICATION_NAME,
-    APPLICATION_COLUMNS.ORGANISATION,
-    APPLICATION_COLUMNS.STAGE,
-    APPLICATION_COLUMNS.REVIEWER_ACTION,
-  ],
-  supervisor: [
+  assigner: [
```
Comment (CarlosNZ) [inline src/utils/data/columnsPerUserRole.ts:24]: Shouldn't "STAGE" still be visible for Assigner?

Comment (CarlosNZ) [reply src/utils/data/columnsPerUserRole.ts:24]: And consolidator?

**Review by CarlosNZ — COMMENTED** [review]
When I'm logged in as one of the Assigners, I'm not seeing any Actions available to me in the "ReviewTest" application list. Should there be?

**Comment by nmadruga** [comment]
> When I'm logged in as one of the Assigners, I'm not seeing any Actions available to me in the "ReviewTest" application list. Should there be?

It depends if this Assigner has permission to Assign to this template. Have they? I'll check now

File: src/utils/data/columnsPerUserRole.ts (hunk 1 above)
Comment (nmadruga) [reply src/utils/data/columnsPerUserRole.ts:24]: Yes, good point. Adding that as well

**Comment by nmadruga** [comment]
> > When I'm logged in as one of the Assigners, I'm not seeing any Actions available to me in the "ReviewTest" application list. Should there be?
> 
> It depends if this Assigner has permission to Assign to this template. Have they? I'll check now.
> 
> Checked and they do. Great catch, I guess I left the Assigner column for actions out of the list? Will fix it now. Thanks!

OK, investigated and the problem is that: we use the result of `reviewerAssignmentJoin` table to count the `assignedCount` and for existing applications of **ReviewTest** those assignment haven't been created, it only create for new applications submitted. : P I can update on the back-end branch to already created ones for Assigner users, so we can see that here.

**Comment by CarlosNZ** [comment]
> I can update on the back-end branch to already created ones for Assigner users, so we can see that here.

Great, can you let me know when this is done?

**Comment by nmadruga** [comment]
> > I can update on the back-end branch to already created ones for Assigner users, so we can see that here.
> 
> Great, can you let me know when this is done?

I have been trying for the past 15 minutes, but it seems more difficult that I thought. I guess for now we can merge this in, and later have an issue to add the proper entities in to mirror the current status for an assigner to see their actions available.

**Comment by CarlosNZ** [comment]
> I have been trying for the past 15 minutes, but it seems more difficult that I thought. I guess for now we can merge this in, and later have an issue to add the proper entities in to mirror the current status for an assigner to see their actions available.

Sure, will approve now.

**Review by CarlosNZ — APPROVED** [review]
LGTM. Thanks :)

**Comment by nmadruga** [comment]
Added issue to enter new reviewerAssignmentJoins here: Add proper reviewerAssignmentJoin entries to match the existing applications on db - to this new issue [back-end-#314](https://github.com/openmsupply/application-manager-server/issues/314)

**Pull request merged**